From 59d8f9a2b5afc91026a29708239e994d6166d677 Mon Sep 17 00:00:00 2001
From: daveyannihilation <daveyannihilation@hotmail.com>
Date: Tue, 20 Sep 2016 23:35:04 +0800
Subject: [PATCH 02/20] Settings: Expose condition card colors

PS2 Change to the actual colors in fwb that these attrs call to.

@nathanchance edit: updated for Nougat 7.1

Change-Id: I8de2e2a4f79a28c0fe1f025b4d23937931fe293a
---
 res/layout/condition_card.xml | 16 ++++++++--------
 res/values/projekt_colors.xml |  9 +++++++++
 2 files changed, 17 insertions(+), 8 deletions(-)

diff --git a/res/layout/condition_card.xml b/res/layout/condition_card.xml
index 7c9e46da00..a5eb6c85cf 100644
--- a/res/layout/condition_card.xml
+++ b/res/layout/condition_card.xml
@@ -25,7 +25,7 @@
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:orientation="vertical"
-        android:background="?android:attr/colorAccent"
+        android:background="@color/condition_card_background_color"
         android:elevation="2dp"
         android:clickable="true"
         android:focusable="true">
@@ -44,7 +44,7 @@
                 android:layout_height="wrap_content"
                 android:layout_marginStart="16dp"
                 android:layout_marginEnd="32dp"
-                android:tint="?android:attr/textColorPrimaryInverse" />
+                android:tint="@color/condition_card_icon_color" />
 
             <TextView
                 android:id="@android:id/title"
@@ -52,14 +52,14 @@
                 android:layout_height="wrap_content"
                 android:layout_weight="1"
                 android:textAppearance="?android:attr/textAppearanceMedium"
-                android:textColor="?android:attr/textColorPrimaryInverse" />
+                android:textColor="@color/condition_card_title_text_color" />
 
             <ImageView
                 android:id="@+id/expand_indicator"
                 android:layout_width="wrap_content"
                 android:layout_height="match_parent"
                 android:padding="16dp"
-                android:tint="?android:attr/textColorPrimaryInverse"/>
+                android:tint="@color/condition_card_expander_color" />
 
         </LinearLayout>
 
@@ -81,14 +81,14 @@
                 android:paddingBottom="16dp"
                 android:textAppearance="?android:attr/textAppearanceListItemSecondary"
                 android:alpha=".7"
-                android:textColor="?android:attr/textColorPrimaryInverse" />
+                android:textColor="@color/condition_card_summary_text_color" />
 
             <!-- TODO: Better background -->
             <View
                 android:id="@+id/divider"
                 android:layout_width="match_parent"
                 android:layout_height=".25dp"
-                android:background="@android:color/white" />
+                android:background="@color/condition_card_divider_color" />
 
             <com.android.internal.widget.ButtonBarLayout
                 android:id="@+id/buttonBar"
@@ -107,7 +107,7 @@
                     android:paddingStart="0dp"
                     android:alpha=".8"
                     android:textAlignment="viewStart"
-                    android:textColor="?android:attr/textColorPrimaryInverse"
+                    android:textColor="@color/condition_card_first_button_text_color"
                     style="?android:attr/buttonBarButtonStyle" />
 
                 <Button
@@ -117,7 +117,7 @@
                     android:layout_weight="1"
                     android:alpha=".8"
                     android:textAlignment="viewStart"
-                    android:textColor="?android:attr/textColorPrimaryInverse"
+                    android:textColor="@color/condition_card_second_button_text_color"
                     style="?android:attr/buttonBarButtonStyle" />
 
             </com.android.internal.widget.ButtonBarLayout>
diff --git a/res/values/projekt_colors.xml b/res/values/projekt_colors.xml
index c30d7eda21..ea54a02a24 100644
--- a/res/values/projekt_colors.xml
+++ b/res/values/projekt_colors.xml
@@ -18,4 +18,13 @@
     <!-- Dashboard category and tile background -->
     <color name="dashboard_category_title_background">@color/card_background</color>
     <color name="selectable_card_background">@color/card_background</color>
+    <!-- Condition card colors -->
+    <color name="condition_card_background_color">@*android:color/accent_device_default_light</color>
+    <color name="condition_card_icon_color">@*android:color/primary_text_default_material_dark</color>
+    <color name="condition_card_title_text_color">@*android:color/primary_text_default_material_dark</color>
+    <color name="condition_card_expander_color">@*android:color/primary_text_default_material_dark</color>
+    <color name="condition_card_summary_text_color">@*android:color/primary_text_default_material_dark</color>
+    <color name="condition_card_divider_color">@android:color/white</color>
+    <color name="condition_card_first_button_text_color">@*android:color/primary_text_default_material_dark</color>
+    <color name="condition_card_second_button_text_color">@*android:color/primary_text_default_material_dark</color>
 </resources>
-- 
2.11.1

